module/users/helper.js

```diff
@@ -610,32 +622,77 @@ module.exports = class UsersHelper {
    * @param {String} pageNo - Page number.
    * @param {String} pageSize - Page size.
    * @param {String} searchText - Search text.
+   * @param {String} userId - User Id.
    * @returns {Array} - Get user targeted programs.
    */
 
-  static programs(bodyData, pageNo, pageSize, searchText) {
+  static programs(bodyData, pageNo, pageSize, searchText, userId) {
     return new Promise(async (resolve, reject) => {
       try {
-        let targetedProgrms = await programsHelper.forUserRoleAndLocation(
+       
+        let targetedProgramIds = [];
+        let nonTargetedProgramIds = []
+        let programCount;
+
+        // getting all program details matching the user profile. not passing pageSize and pageNo to get all data.
+         let targetedPrograms = await programsHelper.forUserRoleAndLocation(
           bodyData,
-          pageSize,
-          pageNo,
-          searchText
+          "", 
+          "",
+          searchText,
+          ["_id"]
+        );
+
+        // targetedPrograms.data contain all programIds targeted to current user profile.
+        if ( targetedPrograms.success && targetedPrograms.data && targetedPrograms.data.data.length > 0) {
+          targetedProgramIds = gen.utils.arrayOfObjectToArrayOfObjectId(targetedPrograms.data.data);
+          programCount = targetedPrograms.count;
+        }
+
+        // In case user changed profile after joined a program, we need to find the such program details. (programs not targeted to user profile anymore)
+        let nontargetedJoinedPrograms  = await this.getUserJoinedProgramDetailsWithPreviousProfiles(
+          targetedProgramIds,
+          searchText,
+          userId          
         );
+        
+        if ( nontargetedJoinedPrograms.success && nontargetedJoinedPrograms.data ) {
+          nonTargetedProgramIds = nontargetedJoinedPrograms.data;
+          programCount = programCount + nontargetedJoinedPrograms.count; // update program count
+        }
 
-        if (!targetedProgrms.success) {
+        //find total number of programs related to user
+        let userRelatedPrograms = targetedProgramIds.concat(nonTargetedProgramIds);
+
+        if (!userRelatedPrograms.length > 0) {
           throw {
             message: constants.apiResponses.PROGRAM_NOT_FOUND,
           };
         }
 
-        targetedProgrms.data["description"] =
-          constants.apiResponses.PROGRAM_DESCRIPTION;
+        // get program details. also applying pagination here
+        let userRelatedProgramsData = await programsHelper.userRelatedProgramsDetails(
```

Comment:
@VISHNUDAS-tunerlabs Can this be done using programDocuments function of program helper?

Reply:
@aks30 function returns  program details after adding solutions and resourcesStarted keys to response. logic for those we have in this function